.github/workflows/dev_on_dispatch_deploy_pullrequests.yml

```
@@ -0,0 +1,302 @@
+name: "0 [on_dispatch] Deploy preview environment"
+
+on:
+  workflow_dispatch:
+    inputs:
+      api_version:
+        type: string
+        description: "SHA1 of the backend image built during your CI"
+        required: true
+
+jobs:
+  deploy-preview-env-init:
+    name: 'Init'
+    runs-on: ubuntu-latest
+    outputs: 
+      resource_label: ${{ steps.set_resource_label.outputs.resource_label }}
+    steps:
+      - name: "Set resources label name"
+        id: "set_resource_label"
+        run: |
+          BRANCH_NAME=${{ github.ref_name }}
+          BRANCH_SHA1=$(echo -n "${BRANCH_NAME}" | sha1sum | awk '{print $1}' | cut -c1-7)
+          echo "resource_label=$BRANCH_SHA1" >> $GITHUB_OUTPUT
+
+  deploy-pro-on-firebase-preview-env:
+    name: "[PRO] Deploy PR version for validation pr preview backend"
+    needs: [deploy-preview-env-init]
+    uses: ./.github/workflows/dev_on_workflow_deploy_pro_pr_version_generic.yml
+    permissions:
+      id-token: write
+      contents: read
+    secrets:
+      GCP_EHP_SERVICE_ACCOUNT: ${{ secrets.GCP_EHP_SERVICE_ACCOUNT }}
+      GCP_EHP_WORKLOAD_IDENTITY_PROVIDER: ${{ secrets.GCP_EHP_WORKLOAD_IDENTITY_PROVIDER }}
+    with:
+      ENV: "testing"
+      CHANNEL: ${{ needs.deploy-preview-env-init.outputs.resource_label }}
+      REF: "${{ github.ref }}"
+      CACHE_BUCKET_NAME: "passculture-infra-prod-github-runner-cache"
+      resource_label: ${{ needs.deploy-preview-env-init.outputs.resource_label }}
+
+  deploy-api:
+    name: "Deploy pullrequest"
+    runs-on: ubuntu-latest
+    needs: [deploy-preview-env-init, deploy-pro-on-firebase-preview-env]
+    environment: pullrequest
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - name: "Echo resource label"
+        run: echo "${{ needs.deploy-preview-env-init.outputs.resource_label }}"
+
+      - uses: actions/checkout@v4.2.1
```

```suggestion
      - uses: actions/checkout@v4.2.1
        with:
          fetch-depth: 1
```
Pour gagner un peu de temps ⚡

---

Je me rends compte que 1 est la valeur par défaut cf https://github.com/actions/checkout/tree/v4.2.1/